File: Marmalade/MarmaladeODK/source/android/CallbacksInitOuyaPlugin.cpp
/*
 * Copyright (C) 2012, 2013 OUYA, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "CallbacksInitOuyaPlugin.h"
#include "CallbackSingleton.h"

#include "ODK_internal.h"
#include "s3eEdk.h"
#include "s3eEdk_android.h"
#include "IwDebug.h"

#include <stdio.h>

void CallbacksInitOuyaPlugin::RegisterCallback(s3eCallback callback, s3eCallback* savedCallback, int callbackType)
{
	if (*savedCallback)
	{
		//IwTrace(ODK, ("Unregistering Callback"));

		s3eEdkCallbacksUnRegister(
				S3E_EXT_ODK_HASH,
				S3E_ODK_CALLBACKS_MAX,
				callbackType,
				*savedCallback,
				NULL);
		*savedCallback = NULL;
	}

	*savedCallback = callback;

	//IwTrace(ODK, ("Registering Callback"));

	s3eEdkCallbacksRegister(
			S3E_EXT_ODK_HASH,
			S3E_ODK_CALLBACKS_MAX,
			callbackType,
			*savedCallback,
			NULL,
			S3E_FALSE);
}

void CallbacksInitOuyaPlugin::RegisterCallbacks(s3eCallback onSuccess, s3eCallback onFailure)
{
	RegisterCallback(onSuccess, &m_onSuccess, S3E_ODK_CALLBACKS_INIT_OUYA_PLUGIN_ON_SUCCESS);
	RegisterCallback(onFailure, &m_onFailure, S3E_ODK_CALLBACKS_INIT_OUYA_PLUGIN_ON_FAILURE);
}

void CallbacksInitOuyaPlugin::OnSuccess()
{
	s3eInitOuyaPluginSuccessEvent event;

	m_dataInitOuyaPluginSuccessEvent = event; //don't send a temp pointer
	s3eEdkCallbacksEnqueue(S3E_EXT_ODK_HASH, S3E_ODK_CALLBACKS_INIT_OUYA_PLUGIN_ON_SUCCESS, &m_dataInitOuyaPluginSuccessEvent);
}

void CallbacksInitOuyaPlugin::OnFailure(int errorCode, const std::string& errorMessage)
{
	char buffer[256];
	sprintf(buffer, "OnFailure errorCode=%d errorMessage=%s", errorCode, errorMessage.c_str());
	IwTrace(ODK, (buffer));

	s3eInitOuyaPluginFailureEvent event;
	event.m_errorCode = errorCode;
	sprintf(event.m_errorMessage, "%s", errorMessage.c_str());

	m_dataInitOuyaPluginFailureEvent = event; //don't send a temp pointer
	s3eEdkCallbacksEnqueue(S3E_EXT_ODK_HASH, S3E_ODK_CALLBACKS_INIT_OUYA_PLUGIN_ON_FAILURE, &m_dataInitOuyaPluginFailureEvent);
}

File: Corona/InAppPurchases/Corona/callbacksInitOuyaPlugin.lua
-- Copyright (C) 2012, 2013 OUYA, Inc.
--
-- Licensed under the Apache License, Version 2.0 (the "License");
-- you may not use this file except in compliance with the License.
-- You may obtain a copy of the License at
--
--    http://www.apache.org/licenses/LICENSE-2.0
--
-- Unless required by applicable law or agreed to in writing, software
-- distributed under the License is distributed on an "AS IS" BASIS,
-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-- See the License for the specific language governing permissions and
-- limitations under the License.

-----------------------------------------------------------------------------------------
--
-- callbacksRequestInitOuyaPlugin.lua
--
-----------------------------------------------------------------------------------------

globals = require "globals"
json = require "json"

local callbacksInitOuyaPlugin = {}

callbacksInitOuyaPlugin.onSuccess = function ()
	globals.txtStatus.text = "onSuccessRequestInitOuyaPlugin";
end

callbacksInitOuyaPlugin.onFailure = function (errorCode, errorMessage)
	globals.txtStatus.text = "onFailureInitOuyaPlugin";
	if errorCode == nil then
        print("onFailureRequestInitOuyaPlugin: errorCode=(nil)");
	else
        print("onFailureRequestInitOuyaPlugin: errorCode=" .. errorCode);
	end
	
	if errorMessage == nil then
        print("onFailureRequestInitOuyaPlugin: errorMessage=(nil)");
	else
        print("onFailureRequestInitOuyaPlugin: errorMessage=" .. errorMessage);
	end
end

return callbacksInitOuyaPlugin;